tower/src/limit/concurrency/layer.rs

```diff
@@ -22,3 +25,27 @@ impl<S> Layer<S> for ConcurrencyLimitLayer {
         ConcurrencyLimit::new(service, self.max)
     }
 }
+
+/// Enforces a limit on the concurrent number of requests the underlying
+/// service can handle.
+/// This variant accepts a owned semaphore (`Arc<Semaphore>`) which can
+/// be reused across services.
+#[derive(Debug, Clone)]
+pub struct GlobalConcurrencyLimitLayer {
+    semaphore: Arc<Semaphore>,
+}
+
+impl GlobalConcurrencyLimitLayer {
+    /// Create a new `GlobalConcurrencyLimitLayer`.
+    pub fn new(semaphore: Arc<Semaphore>) -> Self {
+        GlobalConcurrencyLimitLayer { semaphore }
+    }
```

Comment:
I'm not *totally* sure why this has to take an `Arc<Semaphore>`...couldn't it alternatively just have a constructor that takes the maximum number of concurrent requests, like `ConcurrencyLimitLayer`? Then, instead of cloning the semaphore and constructing a new `ConcurrencyLimitLayer` for each service, you could just clone the concurrency limit layer.

The advantage of this is that it doesn't put the `tokio::sync` type in the public API of `GlobalConcurrencyLimitLayer`. I feel like ideally, the semaphore should be just an implementation detail, not part of the public API. I can't immediately think of a reason we would ever want to change this code not to use `tokio::sync::Semaphore`, but it feels a little better not to expose it publicly, IMO...